svc-aggregation/agmodel/system.go

```diff
@@ -709,3 +709,56 @@ func GetAllKeysFromTable(table string) ([]string, error) {
 	}
 	return keysArray, nil
 }
+
+//AddElementsToAggregate add elements to the aggregate
+func AddElementsToAggregate(aggregate Aggregate, aggregateURL string) *errors.Error {
+	agg, err := GetAggregate(aggregateURL)
+	if err != nil {
+		return err
+	}
+	aggregate.Elements = append(aggregate.Elements, agg.Elements...)
+
+	if err := DeleteAggregate(aggregateURL); err != nil {
+		return err
+	}
+	if err := CreateAggregate(aggregate, aggregateURL); err != nil {
+		return err
+	}
+	return nil
+}
+
+//RemoveElementsFromAggregate remove elements from an aggregate
+func RemoveElementsFromAggregate(aggregate Aggregate, aggregateURL string) *errors.Error {
+	agg, err := GetAggregate(aggregateURL)
+	if err != nil {
+		return err
+	}
+	aggregate.Elements = removeElements(aggregate.Elements, agg.Elements)
+	if err := DeleteAggregate(aggregateURL); err != nil {
```

Comment:
why there is a delete and create there can be new function which can update existing aggregate 2 db calls can be avoided

Reply:
updated